stac_api/runtime/src/core.py

```diff
@@ -0,0 +1,110 @@
+"""CoreCrudClient extensions for the VEDA STAC API."""
+from datetime import datetime
+from typing import Any, List, Optional, Union
+
+import orjson
+from asyncpg.exceptions import InvalidDatetimeFormatError
+from buildpg import render
+from pydantic import ValidationError
+from pygeofilter.backends.cql2_json import to_cql2
+from pygeofilter.parsers.cql2_text import parse as parse_cql2_text
+
+from fastapi import HTTPException
+from stac_fastapi.pgstac.core import CoreCrudClient
+from stac_fastapi.types.errors import InvalidQueryParameter
+from starlette.requests import Request
+
+from .search import CollectionSearchPost
+
+NumType = Union[float, int]
+
+
+class VedaCrudClient(CoreCrudClient):
+    """Veda STAC API Client."""
+
+    async def _collection_id_search_base(
+        self,
+        search_request: CollectionSearchPost,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Cross catalog search (POST).
+        Called with `POST /search`.
+        Args:
+            search_request: search request parameters.
+        Returns:
+            ItemCollection containing items which match the search criteria.
+        """
+        request: Request = kwargs["request"]
+        pool = request.app.state.readpool
+
+        search_request.conf = search_request.conf or {}
+        req = search_request.json(exclude_none=True, by_alias=True)
+
+        try:
+            async with pool.acquire() as conn:
+                q, p = render(
+                    """
+                    SELECT * FROM collection_id_search(:req::text::jsonb);
+                    """,
+                    req=req,
+                )
+                collections = await conn.fetch(q, *p)
+        except InvalidDatetimeFormatError:
+            raise InvalidQueryParameter(
+                f"Datetime parameter {search_request.datetime} is invalid."
+            )
+
+        return [collection["id"] for collection in collections]
```

Comment:
I will revisit the manual function definition in my local db, but I am currently getting a result without an id key (instead it is the name of the function). `[<Record collection_id_search='noaa-emergency-response'>]`

We can make that work with `return [collection["collection_id_search"] for collection in collections]` but we could also touch up that function to return an object with an id key or I think we can return an array of strings without a key which would probably have to be unpacked from a tuple result as `return [collection for collection in collections]`.